src/test/java/seedu/address/logic/parser/EditLeaveCommandParserTest.java

```diff
@@ -0,0 +1,167 @@
+import static seedu.address.logic.commands.CommandTestUtil.VALID_LEAVE_DESCRIPTION;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_LEAVE_DESCRIPTION_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_LEAVE_END_DATE_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_LEAVE_START_DATE_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_LEAVE_STATUS_APPROVED;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_LEAVE_STATUS_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_LEAVE_TITLE;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_LEAVE_TITLE_DESC;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LEAVE_DATE_START;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LEAVE_STATUS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LEAVE_TITLE;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_LEAVE;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_LEAVE;
+import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD_LEAVE;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.EditLeaveCommand;
+import seedu.address.logic.commands.EditLeaveCommand.EditLeaveDescriptor;
+import seedu.address.model.leave.Date;
+import seedu.address.model.leave.Description;
+import seedu.address.model.leave.Status;
+import seedu.address.model.leave.Title;
+import seedu.address.testutil.EditLeaveDescriptorBuilder;
+
+public class EditLeaveCommandParserTest {
+
+    private static final String MESSAGE_INVALID_FORMAT = String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+            EditLeaveCommand.MESSAGE_USAGE);
+
+    private EditLeaveCommandParser parser = new EditLeaveCommandParser();
+
+    @Test
+    public void parse_missingParts_failure() {
+        assertParseFailure(parser, VALID_LEAVE_STATUS_DESC, MESSAGE_INVALID_FORMAT);
+
+        assertParseFailure(parser, "1", EditLeaveCommand.MESSAGE_NOT_EDITED);
+
+        assertParseFailure(parser, "", MESSAGE_INVALID_FORMAT);
+    }
+
+
+    @Test
+    public void parse_invalidPreamble_failure() {
+        // negative index
+        assertParseFailure(parser, "-5" + VALID_LEAVE_STATUS_DESC, MESSAGE_INVALID_FORMAT);
+
+        // zero index
+        assertParseFailure(parser, "0" + VALID_LEAVE_STATUS_DESC, MESSAGE_INVALID_FORMAT);
+
+        // invalid arguments being parsed as preamble
+        assertParseFailure(parser, "1 some random string", MESSAGE_INVALID_FORMAT);
+
+        // invalid prefix being parsed as preamble
+        assertParseFailure(parser, "1 i/ string", MESSAGE_INVALID_FORMAT);
```

Comment:
do we need a test case of INDEX + Valid prefix + Invalid prefix somewhere? I think the getPreamble only gets the text before the first valid prefix as the preamble.

Reply:
i think the invalid prefix will be parsed as part of the previous tag. Currently address book does not take this as an error. For example, adding i/ after a/ in `add` results in
![Screenshot_2023-11-07_09-49-16](https://github.com/AY2324S1-CS2103T-W11-1/tp/assets/73735221/8cdbfebd-edf7-4001-8f5f-53dfbabe0b0e)

Reply:
oh true, then i think the tests are fine